controller/src/beerocks/master/wfa_ca.cpp

```
@@ -0,0 +1,33 @@
+#include "wfa_ca.h"
+
+#include <beerocks/bcl/beerocks_string_utils.h>
+#include <beerocks/bcl/network/network_utils.h>
+
+#include <beerocks/tlvf/beerocks_message.h>
+#include <beerocks/tlvf/beerocks_message_bml.h>
+
+#include <easylogging++.h>
+
+using namespace beerocks;
+
+void wfa_ca::handle_wfa_ca_message(
+    Socket *sd, std::shared_ptr<beerocks_message::cACTION_HEADER> beerocks_header,
+    ieee1905_1::CmduMessageRx &cmdu_rx, ieee1905_1::CmduMessageTx &cmdu_tx, db &database,
+    task_pool &tasks)
+{
+    auto request = cmdu_rx.addClass<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_REQUEST>();
+
+    if (!request) {
+        LOG(ERROR) << "addClass cACTION_BML_WFA_CA_CONTROLLER_REQUEST failed";
+        return;
```

Can we change the function to return `bool` so the error can also be handled in the future by the caller if necessary?

---

We can, but really there is nothing to be handled.

---

Maybe in the future we will add handling.

---

:-1: to that. I think it's a bad idea to add stuff for "maybe in the future we will".

For error handling, it only makes sense to add a return value that indicates error if the error is not handled yet. The error has been handled here (by logging an error); there is nothing more that can be done by the caller.